src/langchain_google_spanner/vector_store.py

```diff
@@ -0,0 +1,1129 @@
+        if distance_strategy == DistanceStrategy.COSINE:
+            return "COSINE_DISTANCE"
+
+        return "EUCLIDEAN_DISTANCE"
+
+    def getDeleteDocumentsParameters(self, columns) -> Tuple[str, Any]:
+        where_clause_condition = " AND ".join(["{} = @{}".format(column, column) for column in columns])
+
+        param_types_dict =  {column: param_types.STRING for column in columns}
+
+        return where_clause_condition, param_types_dict
+
+    def getDeleteDocumentsValueParameters(self, columns, values) -> Dict[str, Any]:
+        return dict(zip(columns, values))
+
+
+class PGSqlSemnatics(DialectSemantics):
+    """
+    Implementation of dialect semantics for PostgreSQL.
+    """
+
+    def getDistanceFunction(self, distance_strategy=DistanceStrategy.EUCLIDEIAN) -> str:
+        if distance_strategy == DistanceStrategy.COSINE:
+            return "spanner.cosine_distance"
+        return "spanner.euclidean_distance"
+
+    def getDeleteDocumentsParameters(self, columns) -> Tuple[str, Any]:
+        where_clause_condition = " AND ".join(["{} = ${}".format(column, index + 1) for index, column in enumerate(columns)])
+
+        value_placeholder_list = ["p{}".format(index + 1) for index in range(len(columns))]
+
+        param_types_dict =  {value_placeholder: param_types.STRING for value_placeholder in value_placeholder_list}
+
+        return where_clause_condition, param_types_dict
+
+    def getDeleteDocumentsValueParameters(self, columns, values) -> Dict[str, Any]:
+        value_placeholder_list = ["p{}".format(index + 1) for index in range(len(columns))]
+        return dict(zip(value_placeholder_list, values))
+
+
+
+class QueryParameters:
+    """
+    Class representing query parameters for nearest neighbors search.
+    """
+
+    class NearestNeighborsAlgorithm(Enum):
+        """
+        Enum for nearest neighbors search algorithms.
+        """
+
+        BRUTE_FORCE = 1
+
+    def __init__(
+        self,
+        algorithm=NearestNeighborsAlgorithm.BRUTE_FORCE,
+        distance_strategy=DistanceStrategy.EUCLIDEIAN,
+        read_timestamp: Optional[datetime.datetime] = None,
+        min_read_timestamp: Optional[datetime.datetime] = None,
+        max_staleness: Optional[datetime.timedelta] = None,
+        exact_staleness: Optional[datetime.timedelta] = None,
+    ):
+        """
+        Initialize query parameters.
+
+        Parameters:
+        - algorithm (NearestNeighborsAlgorithm): The nearest neighbors search algorithm. Defaults to NearestNeighborsAlgorithm.BRUTE_FORCE.
+        - distance_strategy (DistanceStrategy): The distance calculation strategy. Defaults to DistanceStrategy.EUCLIDEAN.
+        - staleness (int): The staleness value. Defaults to 0.
+        """
+        self.algorithm = algorithm
+        self.distance_strategy = distance_strategy
+
+        key: Optional[str]
+        value: Any
+
+        self.staleness = None
```

Comment:
This doesn't match the fields listed.

Reply:
It's an internal field, which we would be deriving from the staleness type and value passed by the user.